packages/types-codec/src/extended/Map.ts

```diff
@@ -46,7 +46,7 @@ function decodeMapFromMap<K extends Codec, V extends Codec> (registry: Registry,
       output.set(
         key instanceof KeyClass
           ? key
-          : new KeyClass(registry, isComplex ? JSON.parse(key as string) : key),
+          : new KeyClass(registry, isComplex && typeof key === 'string' ? JSON.parse(key as string) : key),
```

Comment:
Weird that `as string` is needed if the typeof check has been done.

Reply:
Yep, it isn't needed